build.sbt

```diff
@@ -105,7 +105,8 @@ lazy val core = (project in file("."))
         List(
           // We use JavaConverters to remain backwards compatible with Scala 2.12,
           // and to avoid a dependency on scala-collection-compat
-          "-Wconf:cat=deprecation:i"
+          "-Wconf:cat=deprecation:i",
+          "-Wconf:cat=unused-nowarn:s"
```

Comment:
Turned off the whole thing for 2.12 and it worked.
Hopefully it would error-out in 2.13 for a reasonable scenario, so we can avoid the negative effect of the work around.

Reply:
My suggestion was/is
```scala
      case Some((2, minor)) if minor == 12 =>
        List(
          // Scala 2.12 doesn't always handle @nowarn correctly,
          // and doesn't seem to like @deprecated case class fields with default values.
          "-Wconf:src=src/main/scala/com/cognite/sdk/scala/v1/assets.scala&cat=deprecation:i"
        )
```
Fairly sure I tested it on my machine yesterday and that it worked after removing the redundant `@nowarn` on the case class itself.

As an alternative we can ignore deprecation warnings entirely on 2.12, if we restrict the ignore to only the necessary files on 2.13 , but we shouldn't turn off deprecation warnings entirely on both 2.12 and 2.13.

Reply:
@wjoel had to change the path to `Wconf:src=src/main/scala/com/cognite/sdk/scala/v1/resources/assets.scala&cat=deprecation:i"` while keeping the `@nowarn` for the case class. Otherwise scala 3 will break.

Reply:
"-Wconf:since=2.1.0&cat=deprecation:i" seems to work on all scala 2.12, 2.13 and 3 for me, you can remove the @nowarn as well 😁